components/Crypto.tsx

```diff
@@ -0,0 +1,194 @@
+  Animated,
+  Image,
+  TouchableOpacity,
+} from 'react-native';
+
+import {useNavigation} from '@react-navigation/native';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import {Crypto} from '../models/types';
+import AnimatedText from './AnimatedText';
+import Colors from '../utils/Colors';
+
+interface Props {
+  crypto: Crypto;
+}
+
+function CryptoDetail({crypto}: Props) {
+  const navigation = useNavigation();
+  const spinAnim = useRef(new Animated.Value(0)).current;
+
+  useEffect(() => {
+    Animated.loop(
+      Animated.timing(spinAnim, {
+        toValue: 1,
+        duration: 1200,
+        useNativeDriver: true,
+      }),
+    ).start();
+  }, [spinAnim]);
+
+  const spin = spinAnim.interpolate({
+    inputRange: [0, 1],
+    outputRange: ['0deg', '360deg'],
+  });
+
+  const handlePress = () => {
+    if (navigation)
+      navigation.navigate('DetailCrypto' as never, {crypto: crypto} as never);
+  };
+
+  var value;
+  var valuePercentage;
+
+  if (crypto.market_data.price_usd) {
+    value = crypto.market_data.price_usd.toLocaleString('en-US', {
+      minimumFractionDigits: 2,
+      maximumFractionDigits: 2,
+    });
+  }
+
+  if (crypto.market_data.percent_change_usd_last_1_hour) {
+    valuePercentage =
+      crypto.market_data.percent_change_usd_last_1_hour.toLocaleString(
+        'en-US',
+        {
+          minimumFractionDigits: 2,
+          maximumFractionDigits: 2,
+        },
+      );
+  }
+
+  return (
+    <>
+      <View style={styles.container}>
+        <TouchableOpacity
+          style={styles.touchableContainer}
+          onPress={handlePress}>
+          {!Object.keys(crypto).length ? (
+            <Animated.View
+              style={[styles.loaderContainer, {transform: [{rotate: spin}]}]}>
+              <Icon name="autorenew" size={35} color="#385775" />
+            </Animated.View>
+          ) : (
+            <>
+              <View style={[styles.flexLogo, styles.leftAlign]}>
+                <Image source={{uri: crypto.image}} style={styles.img} />
+                <View style={[styles.insideContainer, styles.leftAlign]}>
+                  <Text style={styles.title}>{crypto.name}</Text>
+                  <Text style={styles.subTitle}>{crypto.symbol}</Text>
+                </View>
+              </View>
+              <View style={[styles.insideContainer, styles.rightAlign]}>
+                {value && crypto.market_data.price_usd ? (
+                  <AnimatedText text={value} page="Home"></AnimatedText>
+                ) : (
+                  <Text style={styles.title}>Not available</Text>
+                )}
+                {crypto.market_data.percent_change_usd_last_1_hour &&
+                crypto.market_data.percent_change_usd_last_1_hour > 0 ? (
+                  <View style={styles.iconContainer}>
+                    {valuePercentage &&
+                    crypto.market_data.percent_change_usd_last_1_hour ? (
+                      <AnimatedText
```

Comment:
you can put it like this, if it doesn't have any child: <div />

<AnimatedText
         text={valuePercentage}
         type="Positive" />